web/src/components/njwds-extended/FileInput.test.tsx

```diff
@@ -0,0 +1,85 @@
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+import { FileInput } from "./FileInput";
+
+describe("<FileInput />", () => {
+  let file: File;
+
+  beforeEach(() => {
+    file = new File(["my cool file contents"], "cool.png", { type: "image/png" });
+  });
+
+  it("passes file to onchange handler", async () => {
+    const mockOnChange = jest.fn();
+    render(
+      <FileInput
+        accept={["PNG"]}
+        errorMessageRequired="error-message-required"
+        helperText="input-label"
+        onChange={mockOnChange}
+        value={undefined}
+      />
+    );
+
+    const uploader = screen.getByTestId("file-input");
+
+    fireEvent.change(uploader, { target: { files: [file] } });
+
+    const base64Contents = Buffer.from("my cool file contents", "utf8").toString("base64");
+
+    await waitFor(() => {
+      expect(mockOnChange).toHaveBeenCalledWith({
+        fileType: "PNG",
+        sizeInBytes: file.size,
+        base64Contents,
+        filename: "cool.png",
+      });
+    });
+  });
+
+  it("refuses files larger than the provided max file size", async () => {
+    const mockOnChange = jest.fn();
+    const errorMessageFileSize = "${fileName} exceeds maximum size of ${maxFileSize} MB.";
+
+    file = new File(["x".repeat(1048576 / 2)], "cool.png", { type: "image/png" }); // .5 MB file
+    render(
+      <FileInput
+        accept={["PNG"]}
+        errorMessageRequired="error-message-required"
+        errorMessageFileSize={errorMessageFileSize}
+        helperText="input-label"
+        maxSizeInMegabytes={0.4}
+        onChange={mockOnChange}
+        value={undefined}
+      />
+    );
+
+    const uploader = screen.getByTestId("file-input");
+
+    fireEvent.change(uploader, { target: { files: [file] } });
+
+    expect(mockOnChange).not.toHaveBeenCalled();
+    expect(screen.getByText("cool.png exceeds maximum size of 0.4 MB.")).toBeInTheDocument();
+  });
+
+  it("refuses files of the wrong type", async () => {
+    const mockOnChange = jest.fn();
+    const errorMessageFileType = "${fileName} file format is not supported.";
+    render(
+      <FileInput
+        accept={["PDF"]}
+        errorMessageRequired="error-message-required"
+        errorMessageFileType={errorMessageFileType}
+        helperText="input-label"
+        onChange={mockOnChange}
+        value={undefined}
+      />
+    );
+
+    const uploader = screen.getByTestId("file-input");
+
+    fireEvent.change(uploader, { target: { files: [file] } });
+
+    expect(mockOnChange).not.toHaveBeenCalled();
+    expect(screen.getByText("cool.png file format is not supported.")).toBeInTheDocument();
```

Comment:
If this is in config, we should use the config in this. If content changes the message, then this test will start failing

Reply:
and the one above

Reply:
I provided a default value to the `errorMessageFileSize` and `errorMessageFileType` values because these errors require multiple currently optional params. My thinking was that I was trying to test the defaults which is why I didn't pull from the Config, but that's really not what this test is currently doing.

Maybe I'm over-engineering this. It might not be entirely obvious where the default provided copy is coming from. I was trying to account for an edge case that might not really be a big deal.

Reply:
Is this comment still relevant? We want to test that this value came from config, right? Since there's no default anymore?